### PR TITLE
Add Aurora Postgres cluster module

### PR DESCRIPTION
**Pull request by minaslater**
Ignore README for now, will update before un-drafting

### CODE REVIEW

File: aurora-postgres/parameter-group/main.tf [hunk 1]
```diff
@@ -0,0 +1,31 @@
+locals {
+  version_components = split(".", var.engine_version)
+  postgres_family = join(
+    "",
+    [
+      "postgres",
```
Comment (thoughtbot-github) [inline aurora-postgres/parameter-group/main.tf:6]: Screenshot 2024-04-02 at 4 03 42 PM

Similar comment here, if we want this for aurora maybe we're leaning in and going with an aurora family? 

Very open to debate but it seems inline with a new module.

**Review by thoughtbot-github — APPROVED** [review]
What was the reasoning behind aurora? It feels so similar to the RDS module we have, I'm not against it, I'm just curious what the benefits are.

a few minor comments but LGTM

**Comment by minaslater** [comment]
> What was the reasoning behind aurora? It feels so similar to the RDS module we have, I'm not against it, I'm just curious what the benefits are.

It's similar to the RDS module, probably because I modeled this one after it instead of starting it from scratch. 😅 

A few reasons for using Aurora for Quench:
- [This article](https://medium.com/awesome-cloud/aws-difference-between-amazon-aurora-and-amazon-rds-comparison-aws-aurora-vs-aws-rds-databases-60a69dbec41f) lists some key differences between RDS and Aurora; tbf, it had a clear bias toward Aurora.
- Quench's old inventory app had been using an Aurora cluster, and without a super clear picture of the level of usage during the inventory days, I wanted to go with what already works for them.
- And honestly, we hadn't made a definitive decision between RDS and Aurora; I wanted to spike this module and see if it was easy enough to do. If it took a long time, I would've just gone with what we already had.

File: aurora-postgres/cluster/variables.tf [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+variable "enabled_cloudwatch_logs_exports" {
+  type        = list(string)
+  description = "Set of log types to enable for exporting to CloudWatch logs. If omitted, no logs will be exported"
+  default     = []
+}
+
+variable "engine" {
+  type        = string
+  description = "RDS database engine; defaults to Postgres"
```
[the remark this answers is not in the document]
Comment (minaslater) [reply aurora-postgres/cluster/variables.tf:63]: You're right that is something I missed from porting over the example.

**Pull request closed without merging**